networkx/generators/random_graphs.py

```diff
@@ -1293,3 +1294,47 @@ def my_function(b):
             j = int(math.ceil(n * kernel_root(i / n, j / n, r)))
             graph.add_edge(i - 1, j - 1)
     return graph
+
+
+@py_random_state(2)
+def random_ordered_tree(n, seed=None, directed=False):
+    """
+    Creates a random ordered tree
+
+    Parameters
+    ----------
+    n : int
+        A positive integer representing the number of nodes in the tree.
+
+    seed : integer, random_state, or None (default)
+        Indicator of random number generation state.
+        See :ref:`Randomness<randomness>`.
+
+    directed : bool
+        if the edges are one-way
+
+    Returns
+    -------
+    networkx.OrderedDiGraph | networkx.OrderedGraph
+
+    Example
+    -------
+    >>> import networkx as nx
+    >>> assert len(nx.random_ordered_tree(n=1, seed=0).nodes) == 1
+    >>> assert len(nx.random_ordered_tree(n=2, seed=0).nodes) == 2
+    >>> assert len(nx.random_ordered_tree(n=3, seed=0).nodes) == 3
+    >>> otree = nx.random_ordered_tree(n=5, seed=3, directed=True)
+    >>> print(nx.forest_str(otree))
+    ╙── 0
+        └─╼ 1
+            └─╼ 4
+                ├─╼ 2
+                └─╼ 3
+    """
+    from networkx.utils import create_py_random_state
+
+    rng = create_py_random_state(seed)
```

Comment:
I think the `@py_random_state` decorator already does this, right?

Reply:
I'm personally not a fan of decorators if they don't save you a line. It makes the code easier to statically parse. I can move to the decorator style for the sake of stylistic consistency, but the end result should be functionally equivalent.

Reply:
But you use the decorator just before the function and then redo what the decorator does. 
Just remove these 3 lines and you're good.

Reply:
Ah, I guess I did do that. I actually removed the entire function. I agreed with @rossbar's comment that its best not to expand the API if we are also adding `create_using` to `random_tree`.